ns3_adapter/src/ns3_adapter.cpp

```diff
@@ -267,29 +279,29 @@ void NS3Adapter::pre_spin()
             {
                 std::lock_guard<std::mutex> lock(cfg_mutex_);
             }
-            ROS_INFO_STREAM("Attempting to connect to NS3");
+            ROS_DEBUG("Attempting to connect to NS3");
             boost::system::error_code ec;
            // ROS_INFO("Connecting to %s:%u", cfg.dsrc_address.c_str(), cfg.dsrc_listening_port);
-            ROS_INFO("Local port: %u", ns3_broadcasting_port_);
+            ROS_DEBUG("Remote broadcasting port: %u", ns3_broadcasting_port_);
+            ROS_DEBUG("Remote registration port: %u", ns3_registration_port_);
+
             try {
-                if (!ns3_client_.connect(ns3_address_, ns3_broadcasting_port_,
-                                          ns3_listening_port_, ec))
+                if (!ns3_client_.connect_registration_and_broadcasting(ns3_address_, ns3_broadcasting_port_, ns3_registration_port_,
+                                          ns3_v2x_listening_port_, ns3_time_listening_port_, ec))
                 {
                     ROS_WARN_STREAM("Failed to connect, err: " << ec.message());
                 }
             }catch(std::exception e)
             {
                 ROS_ERROR_STREAM("Exception connecting to CARMA Ambassador: " << e.what() << " error_code: " << ec.message());
             }
-
-
             connecting_ = false;
         }));
     }
 
-    ns3_reg_client_.connect(ns3_address_, ns3_registration_port_);
-    std::string handshake_msg = compose_handshake_msg(vehicle_id_, role_id_, std::to_string(ns3_listening_port_), host_ip_);
-    ROS_WARN_STREAM("handshake_msg: " << handshake_msg);
+    //ns3_client_.connect_registration(ns3_address_, ns3_registration_port_);
+    std::string handshake_msg = compose_handshake_msg(vehicle_id_, role_id_, std::to_string(ns3_v2x_listening_port_), std::to_string(ns3_time_listening_port_), host_ip_);
+    ROS_DEBUG_STREAM("handshake_msg: " << handshake_msg);
     broadcastHandshakemsg(handshake_msg);
```

Comment:
This is in the pre_spin method. Does this get executed on every cycle of the ros node? I think before this to avoid dropping registrations, we would continuously send them indefinitely. Is that still the case?

Reply:
yeah this is called everytime before the spin(), and the node is still sending that handshake message indefinitely. However, it no longer resets the client every node spin, which I thought was unnecessary. It should technically detect it closed and reset itself with `connect_registration_and_broadcasting` function.